### PR TITLE
Enable Authenticating with Azure with Certificates using Azure SDK for Go's Generic NewDefaultAzureCredential

### PR DESCRIPTION
**Pull request by bryan-cox**
This enhancement proposes enabling image registry, ingress, cloud network config, and storage operators(azure-file and azure-disk) to accept authenticating with Azure with certificates using Azure SDK for Go's generic function [NewDefaultAzureCredential](https://github.com/Azure/azure-sdk-for-go/blob/4ebe2fa68c8f9f0a0737d4569810525b4ac45834/sdk/azidentity/default_azure_credential.go#L63).

1. [Ingress - Refactor to use Azure SDK default cred chain](https://github.com/openshift/cluster-ingress-operator/pull/1151)
2. [Image Registry - Refactor to use Azure SDK default cred chain](https://github.com/openshift/cluster-image-registry-operator/pull/1131)
3. [Cluster Network Operator PR](https://github.com/openshift/cluster-network-operator/pull/2522)
4. [Cloud Network Config - Refactor to use Azure SDK default cred chain](https://github.com/openshift/cloud-network-config-controller/pull/156)
5. [Cluster Storage Operator PR](https://github.com/openshift/cluster-storage-operator/pull/517)
6. [CSI Operator - Separate Azure Authentication for Azure File and Disk](https://github.com/openshift/csi-operator/pull/290) - TBD adding the secrets CSI volume and volume mount for Azure file and disk

### CODE REVIEW

File: enhancements/hypershift/enable-azure-creds-via-cert.md [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+### Non-Goals
+
+N/A
+
+## Proposal
+
+We propose updating the Azure API authentication methods in image registry, ingress, cloud network config, and storage 
+operators to use the using Azure SDK for Go's generic function [NewDefaultAzureCredential](https://github.com/Azure/azure-sdk-for-go/blob/4ebe2fa68c8f9f0a0737d4569810525b4ac45834/sdk/azidentity/default_azure_credential.go#L63).
+This function walks through creating a valid Azure credential chain used to authenticate with Azure API based on what
+environment variables are set up.
```
Comment (bennerv) [inline enhancements/hypershift/enable-azure-creds-via-cert.md:64]:
```suggestion
This function walks through a chain of Azure authentication types, using environment variables, Instance Metadata Service (IMDS), or a file on the local filesystem to authenticate with the Azure API.  
```

nit: it's not just environment variable based. It also walks through Managed Identity credentials using IMDS endpoint, a local file on your filesystem for azure-cli credential auth, etc.

**Review by bennerv — COMMENTED** [review]
Also should capture `AZURE_CLIENT_SEND_CERTIFICATE_CHAIN` but you said there's be followup. 

Otherwise lgtm

File: enhancements/hypershift/enable-azure-creds-via-cert.md [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+## Proposal
+
+We propose updating the Azure API authentication methods in image registry, ingress, cloud network config, and storage 
+operators to use the using Azure SDK for Go's generic function [NewDefaultAzureCredential](https://github.com/Azure/azure-sdk-for-go/blob/4ebe2fa68c8f9f0a0737d4569810525b4ac45834/sdk/azidentity/default_azure_credential.go#L63).
+This function walks through a chain of Azure authentication types, using environment variables, Instance Metadata Service (IMDS), or a file on the local filesystem to authenticate with the Azure API.  
+
+HyperShift would pass the following environment variables - AZURE_CLIENT_ID, AZURE_TENANT_ID, and 
+AZURE_CLIENT_CERTIFICATE_PATH - to its deployments of image registry, ingress, cloud network config, and storage 
+operators (azure-file and azure-disk) on the hosted control plane. Each of these components would then pass these  
+variables along to NewDefaultAzureCredential.
```
Comment (flavianmissi) [inline enhancements/hypershift/enable-azure-creds-via-cert.md:68]: When configured with workload identity, the image registry operator will configure the image registry in a similar fashion ([source](https://github.com/openshift/cluster-image-registry-operator/blob/master/pkg/storage/azure/azure.go?plain=1#L480-L490)).
Do we need to consider this when changing the operator code to use these env vars? In other words, do we also want the image registry to authenticate using this proposed mechanism?

**Review by flavianmissi — COMMENTED** [review]
Left a question about the image registry (operand). Looks good otherwise!

File: enhancements/hypershift/enable-azure-creds-via-cert.md (hunk 2 above)
Comment (bryan-cox) [reply enhancements/hypershift/enable-azure-creds-via-cert.md]: Yeah probably so. On ARO HCP, on the HCP side image registry will authenticate with the cert method, while the operand would use workload identity.

**Pull request closed without merging**